iModelCore/iModelPlatform/DgnCore/DgnMaterial.cpp

```diff
@@ -13,6 +13,33 @@ namespace dgn_ElementHandler
 
 END_BENTLEY_DGNPLATFORM_NAMESPACE
 
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+void RenderMaterial::_OnLoadedJsonProperties()
+    {
+        BeJsValue materialAssets = GetMaterialAssetsR();
+        if (materialAssets.hasMember("renderMaterial") && materialAssets["renderMaterial"].hasMember("Map"))
+            {
+            BeJsValue map = materialAssets["renderMaterial"]["Map"];
+            BeJsDocument mapToModify;
+            mapToModify.toObject();
+            map.ForEachProperty([&](Utf8CP memberName, BeJsConst memberJson)
+                {
+                if (memberJson.isNumericMember("TextureId")) 
+                    {
+                    // Fix IDs that were previously stored as 64-bit integers rather than as ID strings.
+                    int textureId = memberJson["TextureId"].GetInt();
+                    BeInt64Id id(textureId);
+                    Utf8String hexStr = id.ToHexStr();
+                    mapToModify[memberName]["TextureId"] = hexStr.c_str();
+                    }
+                return false;
+                });
+            if (mapToModify.size() > 0)
+                map.From(mapToModify);

```

Comment:
Doesn't this obliterate any other properties in `map`?

Reply:
Ah yeah.. youre totally right

Reply:
Thanks for catching that, I switched over to your approach of casting to (BeJsValue&) in order to be able to modify it in place.